File: Dec2020/6Dec/StringMatchingInAnArray.java
/*
Problem Name: 1408. String Matching in an Array
Problem Link: https://leetcode.com/problems/string-matching-in-an-array/
*/
class Solution {
    public List<String> stringMatching(String[] words) {
        List<String> ans = new ArrayList<String>();
        for(int i=0;i<words.length;i++){
            for(int j=0;j<words.length;j++){
                if(i == j)
                    continue;
                if(words[j].indexOf(words[i]) >=0 ){
                    ans.add(words[i]);
                    break;
                }
            }
        }
        return ans;
    }
}

File: Dec2020/6Dec/StringMatchingInAnArray.cpp
/*
Problem name: 1408. String Matching in an Array
Problem link: https://leetcode.com/problems/string-matching-in-an-array/
*/

class Solution {
public:
    vector<string> stringMatching(vector<string>& words) {
        vector<string> res;
        
        for(int i=0;i<words.size();++i){
            for(int j=0;j<words.size();++j){
                if(i!=j && words[i].size() <= words[j].size()){
                    if(words[j].find(words[i]) != -1){
                        res.push_back(words[i]);
                        break;
                    }
                }
            }
        }
        return res;
    }
};
